src/main/java/org/wildfly/naming/client/WildFlyRootContext.java

```diff
@@ -97,58 +100,90 @@ public NameParser getNativeNameParser() {
         return NAME_PARSER;
     }
 
-    protected Object lookupNative(final Name name) throws NamingException {
+    @Override
+    public Object lookup(String name) throws NamingException {
```

Comment:
No problem with this, but it's probably not necessary since AbstractContext#lookup(String) already delegates to the NameParser which in this case is CompositeName::new.